### DISCUSSION TITLE
How to get nnUNetv2_find_best_configuration to work? (problems with fold)

### DISCUSSION THREAD

**Discussion by eziolanza** [MIC-DKFZ/nnUNet#1677] [Q&A] [answered]
Hello everyone, thanks in advance for the help. 

I trained my dataset like this

`!nnUNetv2_train 1002 2d all`

Then I wanted to use the nnUNetv2_find_best_configuration function to get inference commands. However, I get an error declaring that I have not trained fold 0. It does not recognize the "all" type of training.  Do I have to repeat 5 different trainings, one for each fold? Or at least train fold 0 to use the nnUNetv2_find_best_configuration  ?

Thanks for your help.

**Comment by vcasellesb** [comment, marked as answer]
Hi eziolanza. 

I think nnUNetv2_find_best_configuration requires you to have trained more than one fold and/or more than one configuration. You trained one configuration and one fold -- 'all' doesn't mean nnU-Net will train all folds, rather than nnU-Net will use **all training images** for training (effectively eliminating any possible real folds) -- so there are is no best configuration given that there is only *one* configuration. I hope I have made sense.

Vicent
